Modules/ITS/src/ITSThresholdCalibrationTask.cxx

```diff
@@ -0,0 +1,456 @@
+#include "ITS/ITSThresholdCalibrationTask.h"
+#include <DataFormatsITS/TrackITS.h>
+#include <DataFormatsITSMFT/ROFRecord.h>
+#include <Framework/InputRecord.h>
+#include "ReconstructionDataFormats/Vertex.h"
+#include "ReconstructionDataFormats/PrimaryVertex.h"
+#include "TLine.h"
+#include "TLatex.h"
+#include <Framework/DataSpecUtils.h>
+
+using namespace o2::itsmft;
+using namespace o2::its;
+
+namespace o2::quality_control_modules::its
+{
+
+ITSThresholdCalibrationTask::ITSThresholdCalibrationTask() : TaskInterface()
+{
+}
+
+ITSThresholdCalibrationTask::~ITSThresholdCalibrationTask()
+{
+  delete hSuccessRate;
+
+  for (int iScan = 0; iScan < 3; iScan++) {
+    for (int iLayer = 0; iLayer < 7; iLayer++) {
+      delete hCalibrationLayer[iLayer][iScan];
+      delete hCalibrationRMSLayer[iLayer][iScan];
+      if (iScan == 2) {
+        delete hCalibrationThrNoiseLayer[iLayer];
+        delete hCalibrationThrNoiseRMSLayer[iLayer];
+      }
+    }
+    for (int iBarrel = 0; iBarrel < 3; iBarrel++) {
+      delete hCalibrationChipSum[iScan][iBarrel];
+      delete hCalibrationChipAverage[iScan][iBarrel];
+      delete hCalibrationChipCounts[iScan][iBarrel];
+
+      delete hCalibrationRMSChipSum[iScan][iBarrel];
+      delete hCalibrationRMSChipAverage[iScan][iBarrel];
+    }
+  }
+
+  for (int iBarrel = 0; iBarrel < 3; iBarrel++) {
+
+    delete hCalibrationThrNoiseChipSum[iBarrel];
+    delete hCalibrationThrNoiseChipAverage[iBarrel];
+
+    delete hCalibrationThrNoiseRMSChipSum[iBarrel];
+    delete hCalibrationThrNoiseRMSChipAverage[iBarrel];
+  }
+}
+
+void ITSThresholdCalibrationTask::initialize(o2::framework::InitContext& /*ctx*/)
+{
+
+  ILOG(Info, Support) << "initialize ITSThresholdCalibrationTask" << ENDM;
+
+  mRunNumberPath = mCustomParameters["runNumberPath"];
+
+  createAllHistos();
+  publishHistos();
+}
+
+void ITSThresholdCalibrationTask::startOfActivity(Activity& /*activity*/)
+{
+  ILOG(Info, Support) << "startOfActivity" << ENDM;
+}
+
+void ITSThresholdCalibrationTask::startOfCycle()
+{
+  ILOG(Info, Support) << "startOfCycle" << ENDM;
+}
+
+void ITSThresholdCalibrationTask::monitorData(o2::framework::ProcessingContext& ctx)
+{
+
+  ILOG(Info, Support) << "START DOING QC General" << ENDM;
+  const auto tunString = ctx.inputs().get<gsl::span<char>>("tunestring");
+  const auto runType = ctx.inputs().get<short int>("runtype");
+  const auto scanType = ctx.inputs().get<char>("scantype");
+
+  string inString(tunString.begin(), tunString.end());
+
+  Int_t iScan;
+  Double_t CalibrationValue;
+  if (scanType == 'V')
+    iScan = 0;
+  else if (scanType == 'I')
+    iScan = 1;
+  else if (scanType == 'T')
+    iScan = 2;
+
+  auto splitRes = split(inString, "Stave:");
+
+  for (auto StaveStr : splitRes) {
+    if (StaveStr.size() > 0) {
+      CalibrationResStruct result = CalibrationParser(StaveStr);
+
+      int currentStave = StaveBoundary[result.Layer] + result.Stave + 1;
+      int currentChip = result.ChipID + 1;
```

Comment:
If I'm not mistaken this will work only for IB when using this number to fill the histo. For IB this number goes from 1 to 9 which corresponds to the binning of the x axis of the corresponding histos, while for MB and OB you have to apply some convertion because you need a number from 1 to 112 (for MB) and from 1 to 196 (for OB). In my offline code I use the following but feel free to change (here I use already your naming for variables):
```
if(layer<5){//ML
        int chipbin = result.ChipID<7 ? (result.ChipID+1)+14*(result.HIC-1)+result.Hs*56 : result.ChipID+14*(result.HIC-1)+result.Hs*56; // this is already from 1 to 112
}
else { //OL
    int chipbin = result.ChipID<7 ? (result.ChipID+1)+14*(result.HIC-1)+result.Hs*98 : result.ChipID+14*(result.HIC-1)+result.Hs*98; //this is already from 1 to 196
} 
```